File: Java_Codes/6. Collections Framework/DequeExample.java
package Collections;

import java.util.ArrayDeque;
import java.util.Deque;

public class DequeExample {

    // Function to demonstrate deque operations
    public static void dequeOperations() {
        // Create a deque using ArrayDeque
        Deque<String> deque = new ArrayDeque<>();

        // Add elements to the front and back of the deque
        deque.addFirst("First");
        deque.addLast("Second");
        deque.addLast("Third");
        deque.addFirst("Zeroth");

        // Display the deque
        System.out.println("Deque: " + deque);

        // Remove elements from the front and back of the deque
        String removedFirst = deque.removeFirst();
        String removedLast = deque.removeLast();
        System.out.println("Removed First Element: " + removedFirst);
        System.out.println("Removed Last Element: " + removedLast);

        // Display the deque after removals
        System.out.println("Deque after removals: " + deque);

        // Peek at the front and back of the deque
        String frontElement = deque.peekFirst();
        String backElement = deque.peekLast();
        System.out.println("Front Element: " + frontElement);
        System.out.println("Back Element: " + backElement);

        // Check if the deque is empty
        boolean isEmpty = deque.isEmpty();
        System.out.println("Is the deque empty? " + isEmpty);
    }

    public static void main(String[] args) {
        // Call the function to demonstrate deque operations
        dequeOperations();
    }
}

File: Projects/PythonProject/Collections_Framework/Collections/DequeExample.py
from collections import deque

def deque_operations():
    # Create a deque
    dq = deque()

    # Add elements to the front and back of the deque
    dq.appendleft("First")
    dq.append("Second")
    dq.append("Third")
    dq.appendleft("Zeroth")

    # Display the deque
    print("Deque:", list(dq))

    # Remove elements from the front and back of the deque
    removed_first = dq.popleft()
    removed_last = dq.pop()
    print("Removed First Element:", removed_first)
    print("Removed Last Element:", removed_last)

    # Display the deque after removals
    print("Deque after removals:", list(dq))

    # Peek at the front and back of the deque
    front_element = dq[0] if dq else None
    back_element = dq[-1] if dq else None
    print("Front Element:", front_element)
    print("Back Element:", back_element)

    # Check if the deque is empty
    is_empty = len(dq) == 0
    print("Is the deque empty?", is_empty)


if __name__ == "__main__":
    deque_operations()
